src/main/java/ca/gc/cyber/ops/assemblyline/java/client/clients/ProxyProperties.java

```diff
@@ -0,0 +1,19 @@
+package ca.gc.cyber.ops.assemblyline.java.client.clients;
+
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Data
+@ConfigurationProperties(prefix = "assemblyline-java-client.proxy")
+public class ProxyProperties {
+
+    /**
+     * Proxy host.
+     */
+    private String host;
+
+    /**
+     * Proxy port.
+     */
+    private String port;
```

Comment:
I made the port an int and added @Min(1). 

Since these properties are optional, they should be checked only when set. Using @Validated would validate them at all times. The intent is to make sure the port has a valid value when it is set. The rest of the logic I think belongs when it is being used (ie config proxy if host is set, etc).